tests/sentry/api/endpoints/release_thresholds/test_release_threshold_status.py

```diff
@@ -405,3 +425,28 @@ def test_get_success_project_slug_filter(self):
         # release2
         r2_keys = [k for k, v in data.items() if k.split("-")[1] == self.release2.version]
         assert len(r2_keys) == 0
+
+    def test_new_issue_count(self) -> None:
```

Comment:
Health checks are part of the API though, so it would be wrong to remove this test. If we remove the logic of ensuring new_issue count health check is calculated and returned, we would not be able to catch the regression.

Reply:
Sure, but we should ensure our unit tests are covering the individual units themselves and not dependent on other units.

so if i have an `API` and an `test_API`
and within the `API` i have a `health_check` helper with a `test_health_check`

the `test_API` should _not_ be validating whether the `health_check` is performing as expected. That's what the `test_health_check` should be validating.

So - i'm suggesting we break this up into separate units.

the `APITest`, the `fetch_issue_counts`, and the `is_issue_counts_healthy`
with tests associated for each of them